www/layouts/partials/new_course_cards.html

```diff
@@ -4,37 +4,43 @@
 {{ $courseStarterSlug := getenv "OCW_IMPORT_STARTER_SLUG" }}
 {{ $typeQuery := querify "type" $courseStarterSlug }}
 {{ if and $studioBaseUrl $courseStarterSlug }}
-{{  with getJSON (print (strings.TrimSuffix "/" $studioBaseUrl) "/api/websites/?" $typeQuery)  }}
-  {{ $results := .results }}
-  {{ if $results }}
-  {{ $breakdowns := (dict "xs-sm" (dict "size" 1 "class" "d-flex d-md-none") "md" (dict "size" 2 "class" "d-none d-md-flex d-lg-none") "lg" (dict "size" 3 "class" "d-none d-lg-flex d-xl-none") "xl" (dict "size" 4 "class" "d-none d-xl-flex")) }}
-  <!-- list adapted from https://getbootstrap.com/docs/4.0/utilities/display/#hiding-elements -->
-  {{ range $breakpoint, $carouselInfo := $breakdowns }}
-    {{ $itemsInCarousel := index $carouselInfo "size" }}
-    {{ $breakpointVisibilityClass := index $carouselInfo "class" }}
-    {{ $isMobile := (eq $itemsInCarousel 1) }}
-    {{ $carouselId := (printf "new-course-carousel-%v" $breakpoint) }}
-    <div class="{{ $breakpointVisibilityClass }}">
-      <div id="{{ $carouselId }}" class="carousel slide" data-interval="false" data-touch="true" data-ride="carousel">
-        <div class="carousel-header d-flex flex-row justify-content-between font-weight-bold">
-          <h2>New Courses</h2>
-          {{ partial "carousel_controls.html" $carouselId }}
+{{  with resources.GetRemote (print (strings.TrimSuffix "/" $studioBaseUrl) "/api/websites/?" $typeQuery)  }}
```

Comment:
So I did some experimentation with this here in `new_course_cards.html` and here is what I found:

Firstly, Sentry needs to be made available to be called on the frontend.  The way I accomplished this was to modify `sentry.js` to return the `Sentry` instance after init, then assign it to the `window` object on page load: 
`sentry.js`:
```
import * as Sentry from "@sentry/browser"

export const initSentry = () => {
  Sentry.init({
    dsn:
      "https://eee58f41dda54d2b814296e12dced4b7@o48788.ingest.sentry.io/5304953"
  })
  return Sentry
}
```
`index.ts`:
```
import * as Sentry from "@sentry/browser"
...
export interface OCWWindow extends Window {
  $: JQueryStatic
  jQuery: JQueryStatic
  Popper: typeof Popper
  Sentry: typeof Sentry
}
...
$(function() {
  // hacky coming-soon popover
  document.querySelectorAll(".coming-soon").forEach(el => {
    tippy(el, {
      content: "Coming soon!",
      trigger: "click",
      placement: "top"
    })
  })

  window.Sentry = initSentry()
})
```
Secondly, we need a partial to handle logging the errors, so I created `base-theme/layouts/partials/sentry_capture_message.html`:

```
<script type="text/javascript">
  window.addEventListener("load", (event) => {
    window.Sentry.captureMessage("{{ . }}")
    console.error("{{ . }}")
  })
</script>
```

Lastly, in each place where `resources.GetRemote` is used, we need to call our new partial when we want to log the exception.  One thing that I noticed about this function is that `.Err` is only populated when the server actually returns an actual 500 error.  If the resource simply isn't found, `nil` is returned.  In this case, you need to use the `{{ else }}` statement on the `{{ with }}` block itself.  I had thought we might be able to create a partial that can have a closing tag to have some reusability around error handling specific to `resources.GetRemote`, but unfortunately it looks like that's only possible with shortcodes.  So, wherever we use `resources.GetRemote`, we'll have to do something like this (this example was done here in `new_course_cards.html`):

```
...
              {{ $resultsSlice := first 10 $results }}
              {{ range $index, $courseItem := $resultsSlice  }}
                {{- if $courseItem.url_path -}}
                  {{- $url := delimit (slice (strings.TrimSuffix "/" $staticApiBaseUrl) "/" $courseItem.url_path "/data.json") "" -}}
                  {{ with resources.GetRemote $url }}1
                    {{ with .Err }}
                      {{ $errorMessage := printf "New courses carousel on %v failed to fetch %v with error: %v" site.BaseURL $url .Err }}
                      {{ partial "sentry_capture_message.html" $errorMessage }}
                    {{ else }}
                      {{- $courseData := . | unmarshal -}}
                      {{ partial "course_carousel_card.html" (dict "itemsInCarousel" $itemsInCarousel "courseData" $courseData "index" $index "urlPath" $courseItem.url_path "numCourses" (len $resultsSlice))}}
                    {{ end }}
                  {{ else }}
                    {{ $errorMessage := printf "New courses carousel on %v failed to fetch %v (404)" site.BaseURL $url }}
                    {{ partial "sentry_capture_message.html" $errorMessage }}
                  {{ end }}
                {{- end -}}
              {{ end }}
...
```

Reply:
This is a good suggestion! 
Thanks, @gumaerc, I'll update the PR